### ISSUE TITLE
2.0: CBLQueryExpression "and:" and "or:" methods cause syntax errors in Obj-C++

### ISSUE THREAD

**Issue by snej** [couchbase/couchbase-lite-ios#1862]
I tried using an `and:` operation in a query in an Objective-C++ file and got a syntax error:
```
/Couchbase/2.0/Objective-C/Tests/TunesPerfTest.mm:221:67: error: expected expression
                             where: [[ARTIST notNullOrMissing] and: [COMP nullOrMissing]]
                                                                  ^
```
The same thing happens with `or:`.

I think the method names will need to be changed.

**Comment by pasin** [comment]
Maybe andWith: and orWith:?

**Comment by snej** [comment]
`andExpression:` and `orExpression:` would follow the Obj-C idiom of using the parameter type in the selector. But it's your call of course :)

**Comment by pasin** [comment]
Thanks @snej . I tried with `andWith:`, and it feels weird when reading the where clause.  `andExpression:` feels better but seems to be long. I will do more research and see if I can come up with a better name.

**Comment by snej** [comment]
Maybe French? `et:` and `ou:` :)

**Comment by pasin** [comment]
I wished I could use `et:` and `ou:`. I changed them to `andExpression:` and `orExpression:`.

**Comment by pasin** [comment]
Fixed in b2569b39fb44d6e1ccbbf21f755a69001796c74a.
